### PR TITLE
[PAL-471] allow manual glucose entry when recommendManualGlucoseEntry is presented

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/PAL-471

### CODE REVIEW

**Review by Camji55 — APPROVED** [review]

File: Loop/View Controllers/StatusTableViewController.swift [hunk 1]
```diff
@@ -1343,7 +1343,7 @@ final class StatusTableViewController: LoopChartsTableViewController {
             hostingController.navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .cancel, target: navigationWrapper, action: #selector(dismissWithAnimation))
             present(navigationWrapper, animated: true)
         } else {
-            let viewModel = CarbEntryViewModel(delegate: loopManager)
+            let viewModel = CarbEntryViewModel(delegate: loopManager, enableManualGlucoseEntry: .recommendManualGlucoseEntry == statusRowMode)
```
Comment (ps2) [inline Loop/View Controllers/StatusTableViewController.swift]: It's a little indirect to enable the glucose entry view based on `statusRowMode`.  If statusRowMode had a higher priority status available, like bolusing, but we were still in a situation without cgm data, I would expect the manual glucose entry view to still show up.

**Review by ps2 — CHANGES_REQUESTED** [review]
I think the manual glucose entry view toggle should be based on the lack of cgm data more directly.

File: Loop/View Controllers/StatusTableViewController.swift [hunk 2]
```diff
@@ -651,7 +651,7 @@ final class StatusTableViewController: LoopChartsTableViewController {
         case status = 0
     }
 
-    private enum StatusRowMode {
+    private enum StatusRowMode: Equatable {
```
Comment (ps2) [inline Loop/View Controllers/StatusTableViewController.swift]: Is this still needed?

**Review by ps2 — APPROVED** [review]
LGTM! Just maybe a little nit.

File: Loop/View Controllers/StatusTableViewController.swift (hunk 2 above)
Comment (nhamming) [reply Loop/View Controllers/StatusTableViewController.swift]: thanks! good catch

**Pull request merged**